src/Keg.sol

```diff
@@ -64,126 +239,108 @@ contract Keg is LibNote {
+        require(bums.length == rad.length, "Keg/unequal-payees-and-amounts");
         require(bums.length > 0, "Keg/no-bums");
-        uint suds = 0;
-        for (uint i = 0; i < wad.length; i++) {
+        for (uint256 i = 0; i < rad.length; i++) {
             require(bums[i] != address(0), "Keg/no-address-0");
-            mugs[bums[i]] = add(mugs[bums[i]], wad[i]);
-            suds          = add(suds, wad[i]);
-            emit PourBeer(bums[i], wad[i]);
+            vat.move(msg.sender, bums[i], rad[i]);
+            emit PourBeer(bums[i], rad[i]);
         }
-        beer = add(beer, suds);
-        require(vat.dai(address(this)) == mul(beer, RAY), "Keg/pour-not-equal-to-brew");
     }
 
-    // User delegates compensation to another address
-    function pass(address bud) external {
-        require(bud != msg.sender, "Keg/cannot_delegate_to_self");
-        require(pals[bud] == address(0), "Keg/bud-already-has-a-pal");
-        // Remove existing delegate
-        if (buds[msg.sender] != address(0)) yank();
-        // Original addr -> delegated addr
-        buds[msg.sender] = bud;
-        // Delegated addr -> original addr
-        pals[bud] = msg.sender;
-        emit HoldMyBeerBro(msg.sender, bud);
-    }
+    // Credits people with rights to withdraw funds from the pool using a preset flight
+    function pour(bytes32 flight, uint256 rad) external note stoppable {
+        Pint[] memory pints = flights[flight];
 
-    // User revokes delegation
-    function yank() public {
-        require(buds[msg.sender] != address(0), "Keg/no-bud");
-        pals[buds[msg.sender]] = address(0);
-        buds[msg.sender] = address(0);
-        emit ByeFelicia(msg.sender, buds[msg.sender]);
+        require(rad > 0, "Keg/rad-zero");
+        require(pints.length > 0, "Keg/flight-not-set");       // numPints will be empty when not set
+        
+        uint256 suds = 0;
+        for (uint256 i = 0; i < pints.length; i++) {
+            Pint memory pint = pints[i];
+            uint256 sud;
+            if (i == pints.length - 1) {
+                // Add whatevers left over to the last mug to account for rounding errors
+                sud = sub(rad, suds);
+            } else {
+                // Otherwise use the share amount
+                sud = mul(rad, pints[i].share) / WAD;
+            }
+            suds = add(suds, sud);
+            vat.move(msg.sender, pint.mug, sud);
+            emit PourBeer(pint.mug, sud);
+        }
     }
 
-    // User withdraws all funds
-    function chug() external {
-        uint pint = mugs[msg.sender] + mugs[pals[msg.sender]];
-        require(pint != uint256(0), "Keg/too-thirsty-not-enough-beer");
-        beer = sub(beer, pint);
-        mugs[msg.sender] = 0;
-        mugs[pals[msg.sender]] = 0;
+    // Pre-authorize a flight distribution of funds
+    function serve(bytes32 flight, address[] calldata bums, uint256[] calldata shares) external note auth {
```

Comment:
I think `seat` would be a better meme for this, like the bums are getting seated at the table.